src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -143,20 +101,25 @@ def run_python_test(test_target, pytest, python_setup, source_root_config, pex_b
   all_input_digests = [
     sources_digest,
     inits_digest.directory_digest,
-    requirements_pex_response.output_directory_digest,
+    resolved_requirements_pex.directory_digest,
   ]
   merged_input_files = yield Get(
     Digest,
     DirectoriesToMerge,
     DirectoriesToMerge(directories=tuple(all_input_digests)),
   )
 
+  interpreter_search_paths = text_type(create_path_env_var(python_setup.interpreter_search_paths))
   pex_exe_env = {'PATH': interpreter_search_paths}
   # TODO(#6071): merge the two dicts via ** unpacking once we drop Py2.
   pex_exe_env.update(subprocess_encoding_environment.invocation_environment_dict)
 
+  # NB: we use the hardcoded and generic bin name `python`, rather than something dynamic like
+  # `sys.executable`, to ensure that the interpreter may be discovered both locally and in remote
```

Comment:
This still has the issue that the interpreter search paths aren't portable across systems, so the paths we look up locally won't necessarily be the remote ones... Happy to punt on this for now, but maybe by adding a TODO saying "We'll need to hard-code paths from a platform" or something, rather than suggesting that this will work remotely.

Reply:
Sounds good. There are two problems I've found so far with remoting and `PATH`:
1) which bin name to use to run PEX
2) exposing to PEX some form of the `PATH` so that it can itself find an appropriate interpreter.

This comment is supposed to address #1. #2 is addressed by `--ppython-setup-interpreter-search-paths`, which we can configure already remoting to use a specific set of values.

I can reword the comment to clarify this distinction.

Reply:
I think my specific thing would be addressed by a comment along the lines of:
```# TODO: Set --python-setup-interpreter-search-paths differently for the host and target platforms, when we introduce platforms in #7735```